src/lib/cli/progress.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import { stdout as progressLog } from 'single-line-log';
+
+// Progress spinner
+const sprite = {
+	i: 0,
+	sprite: [ '⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏' ],
+	next() {
+		this.i++;
+
+		if ( this.i >= this.sprite.length ) {
+			this.i = 0;
+		}
+
+		return {
+			value: this.sprite[ this.i ],
+			done: false,
+		};
+	},
+};
+
+// Icons for different action states/status
+const marks = {
+	pending: '○',
+	running: chalk.blueBright( sprite.next().value ),
+	success: chalk.green( '✓' ),
```

Comment:
Should we have an extra mark for `skipped`? I guess not all imports will run with search-replace so it may be a good idea to mark that step as `skipped` for those imports.

Reply:
Good idea! Added 👍